src/storage/mvcc/txn.rs

```diff
@@ -12,18 +12,99 @@
+                               row_key: &Key,
+                               mut row: Row,
+                               first_meta: &Meta,
+                               ts: u64)
+                               -> Result<RowValue> {
+        // Check for locks that signal concurrent writes.
+        if let Some(lock) = first_meta.get_lock() {
+            if lock.get_start_ts() <= ts {
+                // There is a pending lock. Client should wait or clean it.
+                return Err(Error::KeyIsLocked {
+                    key: row.get_row_key().to_vec(),
+                    primary: lock.get_primary_key().to_vec(),
+                    ts: lock.get_start_ts(),
+                });
+            }
+        }
+        let mut row_value = RowValue::new();
+        row_value.set_row_key(row.take_row_key());
+        let mut pending_cols = BTreeMap::new();
+        for col in row.get_columns() {
+            pending_cols.insert(col.to_vec(), true);
+        }
+        // Find the latest write below our start timestamp.
+        if let Some(x) = first_meta.iter_items().find(|x| x.get_commit_ts() <= ts) {
+            for col in x.get_columns() {
+                if let Some(_) = pending_cols.remove(col.get_name()) {
+                    if col.get_op() == Op::Put {
+                        let value_key = row_key.append_ts_column(x.get_start_ts(), col.get_name());
+                        if let Some(x) = try!(self.read(&value_key)) {
+                            // TODO: Should we keep the order of columns in RowValue same as Row?
+                            row_value.mut_columns().push(col.get_name().to_vec());
+                            row_value.mut_values().push(x);
+                        }
+                    }
+                }
+            }
+        }
+
+        let mut next = first_meta.next_index();
+        loop {
+            if pending_cols.len() == 0 {
+                break;
+            }
+            let meta = match next {
+                Some(x) => try!(self.load_meta(&row_key, x)),
+                None => break,
+            };
+            if let Some(x) = meta.iter_items().find(|x| x.get_commit_ts() <= ts) {
+                for col in x.get_columns() {
+                    if let Some(_) = pending_cols.remove(col.get_name()) {
+                        if col.get_op() == Op::Put {
+                            let value_key =
+                                row_key.append_ts_column(x.get_start_ts(), col.get_name());
+                            if let Some(x) = try!(self.read(&value_key)) {
+                                // TODO: Should we keep the order of columns in RowValue same as
+                                // Row?
+                                row_value.mut_columns().push(col.get_name().to_vec());
+                                row_value.mut_values().push(x);
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
This block of code is an duplication of the code on line 65, can we unify them in the loop?
like 
```rust
let mut meta = first_meta;
loop {
    if let Some(x) = meta.iter_items().find(|x| x.get_commit_ts() <= ts) {
        ....
    }
    if pending_cols.len() == 0 {
         break;
    }
    next = meta.next_index();
    meta = match next {
        Some(x) => try!(self.load_meta(&row_key, x)),
        None => break,
    };
}